server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -548,12 +548,17 @@ public final Optional<SourceKeepMode> sourceKeepMode() {
     }
 
     @Override
-    public void validate(MappingLookup mappers) {
+    public final void validate(MappingLookup mappers) {
         for (Mapper mapper : this.mappers.values()) {
             mapper.validate(mappers);
+            validateSubField(mapper, mappers);
         }
     }
 
+    protected void validateSubField(Mapper mapper, MappingLookup mappers) {
+        mapper.validate(mappers);
+    }
```

Comment:
I am not sure I follow the need for this. it currently does the same call twice? I would expect that we only need to validate in RootObjectMapper, which takes care of the top-level